include/cnl/constant.h

```diff
@@ -63,7 +63,7 @@ namespace cnl {
 #endif
         }
 
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && defined(_HAS_CXX17) && _HAS_CXX17
```

Comment:
Thanks for the PR. Can you confirm that MSVC doesn't support the standard version check macro and that there isn't [a feature test macro](https://en.cppreference.com/w/User:D41D8CD98F/feature_testing_macros#C.2B.2B17) that could be employed instead. It's generally better practice to use a more tightly-focussed test in cases where a compiler has implemented half of C++17 but not the other half.

Finally, could you provide a more descriptive commit message? This commits don't help anyone who looks through the history to try and understand why the change was made. Consider how `git blame` output might be improved with a more detailed message.

Reply:
Just had the same issue with the latest VS2019. Unfortunately I cannot check on which exact version it was fixed, but I suggest doing something like that:

```cpp
#if defined(_MSC_VER) && _MSC_VER < 1927
```

If someone encounters the issue on a VS version for which that fix won't work. i.e. the interval between VS2017 and some older VS2019, we can make that fix more precise using that info.

Also, it seems there's no feature macros related to this issue.

Reply:
Update: actually, [all msvc version up until 1924 accept](https://gcc.godbolt.org/z/PP5dsj) the current `master`, so the fix must've happened in 1925-1927.

Reply:
Can you confirm whether #638 addresses the compiler error for you?